File: social-service/src/main/java/com/eirsteir/coffeewithme/social/domain/friendship/FriendshipStatusConverter.java
package com.eirsteir.coffeewithme.social.domain.friendship;

import com.eirsteir.coffeewithme.social.domain.RequestStatusConverter;
import java.util.stream.Stream;
import javax.persistence.Converter;

@Converter(
    autoApply = true) // automatically apply the conversion logic to all mapped attributes of a
// FriendRequestStatus type.
public class FriendshipStatusConverter
    implements RequestStatusConverter<FriendshipStatus, Integer> {

  public Integer convertToDatabaseColumn(FriendshipStatus friendshipStatus) {
    return friendshipStatus == null ? null : friendshipStatus.getValue();
  }

  public FriendshipStatus convertToEntityAttribute(Integer value) {
    if (value == null) return null;

    return Stream.of(FriendshipStatus.values())
        .filter(v -> v.getValue().equals(value))
        .findFirst()
        .orElseThrow(IllegalArgumentException::new);
  }
}

File: social-service/src/main/kotlin/com/eirsteir/coffeewithme/social/domain/friendship/FriendshipStatusConverter.kt
package com.eirsteir.coffeewithme.social.domain.friendship

import com.eirsteir.coffeewithme.social.domain.RequestStatusConverter
import java.util.stream.Stream
import javax.persistence.Converter

@Converter(autoApply = true)
class FriendshipStatusConverter : RequestStatusConverter<FriendshipStatus, Int> {

    override fun convertToDatabaseColumn(friendshipStatus: FriendshipStatus) = friendshipStatus.getValue()

    override fun convertToEntityAttribute(value: Int): FriendshipStatus =
        FriendshipStatus.values()
            .filter { v -> v.getValue() == value }
            .stream()
            .findFirst()
            .orElseThrow { IllegalArgumentException() }

}